frontend/src/machines/billings/billingMachine.ts

```diff
@@ -0,0 +1,553 @@
+import { createMachine, assign } from "xstate";
+import { GetMemberListResponse } from "../../utils/fetcher";
+
+type Billing = {
+  id: string;
+  title: string;
+  amount: number;
+};
+
+type BillingDetailMember = {
+  id: string;
+  name: string;
+  amount: number;
+};
+
+type BillingDetail = {
+  id: string;
+  name: string;
+  billAmount: number;
+  chargedMemberId: string;
+  isBillEqually: boolean;
+  members: BillingDetailMember[];
+};
+
+type BillingForm = {
+  title?: string;
+  total?: number;
+  chargedMember?: string;
+  isBillEqual: boolean;
+  members?: GetMemberListResponse["data"];
+};
+
+type FORMMODE = "CREATE" | "EDIT";
+
+interface Context {
+  billings?: Billing[];
```

Comment:
`billings` ini kayaknya gak perlu dibuat undefined ? soalnya dia bisa diisi array kosong aja kalau belum fetching. Kecuali kalau dia object, perlu dibuat undefined / null

Reply:
Hmm, berarti pas tipping state nya dia bakal diisi gin ya:

```
| {
      value: { billingFormIdle: "loadingMembers" };
      context: Context & {
        billings: Billing[];
        billingError: undefined;
        formMode: undefined;
        members: undefined;
        membersError: undefined;
        billingDetail: undefined;
        billingDetailError: undefined;
        billingForm: undefined;
        submitBillingError: undefined;
      };
    }
  | 
```

Instead of :
```
| {
      value: { billingFormIdle: "loadingMembers" };
      context: Context & {
        billings: undefined;
        billingError: undefined;
        formMode: undefined;
        members: undefined;
        membersError: undefined;
        billingDetail: undefined;
        billingDetailError: undefined;
        billingForm: undefined;
        submitBillingError: undefined;
      };
    }
  | 
```

Reply:
Eh bentar, kayanya bisa simply gini deh:
```
| {
      value: { billingFormIdle: "loadingMembers" };
      context: Context & {
        billings: [];
        billingError: undefined;
        formMode: undefined;
        members: undefined;
        membersError: undefined;
        billingDetail: undefined;
        billingDetailError: undefined;
        billingForm: undefined;
        submitBillingError: undefined;
      };
    }
  | 
```

Reply:
yang `billings` jadi gak perlu ditulis ulang, karena gak berubah, udah didefine sama `Context`, jadinya gini : 

```
{
      value: { billingFormIdle: "loadingMembers" };
      context: Context & {
        billingError: undefined;
        formMode: undefined;
        members: undefined;
        membersError: undefined;
        billingDetail: undefined;
        billingDetailError: undefined;
        billingForm: undefined;
        submitBillingError: undefined;
      };
    }
```

Reply:
Hmmm, bener jugaa.
Kalau ga di defined, bakal ikut typing Context nya secara otomatis yak